src/Analyzers/CSharp/Analyzers/UseCollectionExpression/CSharpUseCollectionExpressionForArrayDiagnosticAnalyzer.cs

```diff
@@ -89,7 +89,7 @@ private void AnalyzeArrayInitializerExpression(SyntaxNodeAnalysisContext context
             : initializer;
 
         if (!UseCollectionExpressionHelpers.CanReplaceWithCollectionExpression(
-                semanticModel, arrayCreationExpression, skipVerificationForReplacedNode: false, cancellationToken))
+                semanticModel, arrayCreationExpression, skipVerificationForReplacedNode: true, cancellationToken))
```

Comment:
needed because we are genuinely changing the semantics of the starting node (from an array to a span).